kubernetes/structure_persistent_volume_spec.go

```diff
@@ -1518,6 +1559,28 @@ func patchPersistentVolumeSource(pathPrefix, prefix string, d *schema.ResourceDa
 		}
 	}
 
+	if d.HasChange(prefix + "csi") {
+		oldIn, newIn := d.GetChange(prefix + "csi")
+		oldV, oldOk := oldIn.([]interface{})
+		newV, newOk := newIn.([]interface{})
+
+		if newOk && len(newV) > 0 {
+			if oldOk && len(oldV) > 0 {
```

Comment:
I think we could clean the duplication here by setting 

```go
path := pathPrefix + "/csi"
value := expandCSIPersistentDiskVolumeSource(newV)
```

before the conditional, and then just supply them when constructing AddOperation/ReplaceOperation

Reply:
should i just do this for all other as well? maybe extract to a more general function.
id addressed the changes you mentioned, i can open another PR for a more general refactor